From 19a647b67764a46d7b90a566bc5f60ba936d54f2 Mon Sep 17 00:00:00 2001
From: pdalgd <55156186+pdalgd@users.noreply.github.com>
Date: Fri, 21 Feb 2020 13:24:37 +0100
Subject: [PATCH 04/12] NaN not propagating in DCOMBSSQ and SCOMBSSQ

---
 SRC/dcombssq.f | 2 ++
 SRC/scombssq.f | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/SRC/dcombssq.f b/SRC/dcombssq.f
index 79f6d95c9..980a80e66 100644
--- a/SRC/dcombssq.f
+++ b/SRC/dcombssq.f
@@ -80,6 +80,8 @@
       IF( V1( 1 ).GE.V2( 1 ) ) THEN
          IF( V1( 1 ).NE.ZERO ) THEN
             V1( 2 ) = V1( 2 ) + ( V2( 1 ) / V1( 1 ) )**2 * V2( 2 )
+         ELSE
+            V1( 2 ) = V1( 2 ) + V2( 2 )
          END IF
       ELSE
          V1( 2 ) = V2( 2 ) + ( V1( 1 ) / V2( 1 ) )**2 * V1( 2 )
diff --git a/SRC/scombssq.f b/SRC/scombssq.f
index 76bc0e320..cc51a324b 100644
--- a/SRC/scombssq.f
+++ b/SRC/scombssq.f
@@ -80,6 +80,8 @@
       IF( V1( 1 ).GE.V2( 1 ) ) THEN
          IF( V1( 1 ).NE.ZERO ) THEN
             V1( 2 ) = V1( 2 ) + ( V2( 1 ) / V1( 1 ) )**2 * V2( 2 )
+         ELSE
+            V1( 2 ) = V1( 2 ) + V2( 2 )
          END IF
       ELSE
          V1( 2 ) = V2( 2 ) + ( V1( 1 ) / V2( 1 ) )**2 * V1( 2 )
