From 3f17ea8a00a81492f4e01b16edd2b4f0a5138395 Mon Sep 17 00:00:00 2001
From: Ilhan Polat <ilhanpolat@gmail.com>
Date: Mon, 27 Apr 2020 12:36:59 +0200
Subject: [PATCH 05/10] Fix the loop index scalar in ZHEEQUB.f

---
 SRC/zheequb.f | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SRC/zheequb.f b/SRC/zheequb.f
index d698232e8..7d719f41e 100644
--- a/SRC/zheequb.f
+++ b/SRC/zheequb.f
@@ -271,7 +271,7 @@
          AVG = AVG / N
 
          STD = 0.0D0
-         DO I = N+1, N
+         DO I = N+1, 2*N
             WORK( I ) = S( I-N ) * WORK( I-N ) - AVG
          END DO
          CALL ZLASSQ( N, WORK( N+1 ), 1, SCALE, SUMSQ )
