From e69f944b6b4df8f8dd0ef4de6574ec201b415af3 Mon Sep 17 00:00:00 2001
From: Ilhan Polat <ilhanpolat@gmail.com>
Date: Mon, 27 Apr 2020 12:36:59 +0200
Subject: [PATCH 5/6] Fix the loop index scalar in ZHEEQUB.f

---
 SRC/zheequb.f | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SRC/zheequb.f b/SRC/zheequb.f
index d698232e..7d719f41 100644
--- a/SRC/zheequb.f
+++ b/SRC/zheequb.f
@@ -271,7 +271,7 @@
          AVG = AVG / N
 
          STD = 0.0D0
-         DO I = N+1, N
+         DO I = N+1, 2*N
             WORK( I ) = S( I-N ) * WORK( I-N ) - AVG
          END DO
          CALL ZLASSQ( N, WORK( N+1 ), 1, SCALE, SUMSQ )
-- 
2.26.2.windows.1

